File: lib/src/app/context/OnboardingContext.js
"use client";
import { createContext, useContext, useState, useEffect } from "react";

const OnboardingContext = createContext();

export function OnboardingProvider({ children }) {
  const [showSuccessModal, setShowSuccessModal] = useState(false);
  const [newRestaurantId, setNewRestaurantId] = useState(null);

  // context/OnboardingContext.js
  // ... inside your Provider state
  const [trialDaysLeft, setTrialDaysLeft] = useState(7);

  useEffect(() => {
    // Logic to calculate days: 7 - (Current Date - Created Date)
    // For now, let's assume it's stored in localStorage or comes from your User API
    const signupDate = localStorage.getItem("signupDate");
    if (signupDate) {
      const diff = Math.floor((new Date() - new Date(signupDate)) / (1000 * 60 * 60 * 24));
      setTrialDaysLeft(Math.max(0, 7 - diff));
    }
  }, []);

  // Trigger the popup
  const triggerSuccess = (id) => {
    setNewRestaurantId(id);
    setShowSuccessModal(true);
  };

  const closeSuccess = () => {
    setShowSuccessModal(false);
  };

  return (
    <OnboardingContext.Provider value={{ showSuccessModal, newRestaurantId, triggerSuccess, closeSuccess,trialDaysLeft, setTrialDaysLeft }}>
      {children}
    </OnboardingContext.Provider>
  );
}

export const useOnboarding = () => useContext(OnboardingContext);

File: src/app/context/OnboardingContext.dart
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:shared_preferences/shared_preferences.dart';

class OnboardingState {
  final bool showSuccessModal;
  final String? newRestaurantId;
  final int trialDaysLeft;

  OnboardingState({
    required this.showSuccessModal,
    this.newRestaurantId,
    required this.trialDaysLeft,
  });

  OnboardingState copyWith({
    bool? showSuccessModal,
    String? newRestaurantId,
    int? trialDaysLeft,
  }) {
    return OnboardingState(
      showSuccessModal: showSuccessModal ?? this.showSuccessModal,
      newRestaurantId: newRestaurantId ?? this.newRestaurantId,
      trialDaysLeft: trialDaysLeft ?? this.trialDaysLeft,
    );
  }
}

class OnboardingNotifier extends Notifier<OnboardingState> {
  @override
  OnboardingState build() {
    _loadTrialDays();
    return OnboardingState(showSuccessModal: false, trialDaysLeft: 7);
  }

  Future<void> _loadTrialDays() async {
    final prefs = await SharedPreferences.getInstance();
    final signupDateStr = prefs.getString('signupDate');
    if (signupDateStr != null) {
      final signupDate = DateTime.parse(signupDateStr);
      final diff = DateTime.now().difference(signupDate).inDays;
      final daysLeft = 7 - diff;
      state = state.copyWith(trialDaysLeft: daysLeft > 0 ? daysLeft : 0);
    }
  }

  void triggerSuccess(String id) {
    state = state.copyWith(newRestaurantId: id, showSuccessModal: true);
  }

  void closeSuccess() {
    state = state.copyWith(showSuccessModal: false);
  }

  void setTrialDaysLeft(int days) {
    state = state.copyWith(trialDaysLeft: days);
  }
}

final onboardingProvider = NotifierProvider<OnboardingNotifier, OnboardingState>(OnboardingNotifier.new);
